pyvista/core/_validation/validate.py

```diff
@@ -221,28 +392,33 @@ def validate_array(
 
         A copy may also be made to satisfy ``dtype_out`` requirements.
 
-    to_list : bool, default: False
-        Return the validated array as a ``list`` or nested ``list``. Scalar
-        values are always returned as a ``Number``  (i.e. ``int`` or ``float``).
-        Has no effect if ``to_tuple=True``.
+    return_type : str | type, optional
+        Control the return type of the array. The array may be copied,
+        but only if necessary. Must be one of:
+
+        * ``"numpy"`` or ``np.ndarray``
+        * ``"list"`` or ``list``
+        * ``"tuple"`` or ``tuple``
 
-    to_tuple : bool, default: False
-        Return the validated array as a ``tuple`` or nested ``tuple``. Scalar
-        values are always returned as a ``Number``  (i.e. ``int`` or ``float``).
+        .. note::
+
+            For scalar inputs, setting the output type to ``list`` or
+            ``tuple`` will return a scalar, and not an actual ``list``
+            or ``tuple`` object.
 
     name : str, default: "Array"
         Variable name to use in the error messages if any of the
         validation checks fail.
 
     Returns
     -------
-    array_like
+    float | int | bool | Array[float] | Array[int] | Array[bool]
         Validated array. Returned object is:
 
         * an instance of ``np.ndarray`` (default), or
         * a nested ``list`` (if ``to_list=True``), or
```

Comment:
Text needs to be updated to match new generic ArrayLike[T] -> ArrayLike[T] behaviour.

Also should add a new `shape_changed` or `return_reshaped_flag` (or similar) kwarg.
Initially suggested by @darikg to provide similar functionality provided by `_coerce_pointslike_arg` and document the return behaviour here.

Reply:
The returns section has been uddated, and a new `get_flags` param has been added with info about the change in shape